src/components/files/PreViewer.tsx

```diff
@@ -25,15 +28,55 @@ export const PreViewer = ({ s3ObjectId, s3Key }: Props) => {
   const viewableRows = allRows.slice(0, 1000);
 
   return (
-    <>
+    <div className='space-y-4'>
       {viewableRows.length > 1000 && (
-        <div className='mb-3 w-full border bg-amber-100 p-2 text-amber-700'>
-          Only showing the first 1000 rows
+        <div
+          className={classNames(
+            'flex items-center gap-2 rounded-lg p-3',
+            'bg-amber-50 dark:bg-amber-900/30',
+            'border border-amber-200 dark:border-amber-700',
+            'text-amber-800 dark:text-amber-200'
+          )}
+        >
+          <ExclamationTriangleIcon className='h-5 w-5 flex-shrink-0' />
+          <p className='text-sm font-medium'>Only showing the first 1,000 rows for performance</p>
         </div>
       )}
-      <pre className='border-round-xs m-0 mt-4 inline-block w-full overflow-auto border border-solid border-current bg-white p-3'>
-        {viewableRows.join('\n')}
-      </pre>
-    </>
+
+      <div
+        className={classNames(
+          'overflow-hidden rounded-lg',
+          'border border-gray-200 dark:border-gray-700',
+          'bg-white dark:bg-gray-900',
+          'shadow-sm'
+        )}
+      >
+        <div
+          className={classNames(
+            'border-b px-4 py-2',
+            'border-gray-200 dark:border-gray-700',
+            'bg-gray-50 dark:bg-gray-800'
+          )}
+        >
+          <div className='flex items-center justify-between'>
+            <h3 className='text-sm font-medium text-gray-900 dark:text-gray-100'>File Preview</h3>
+            <span className='text-xs text-gray-500 dark:text-gray-400'>
+              {s3Key.split('/').pop()}
+            </span>
+          </div>
+        </div>
+
+        <pre
+          className={classNames(
+            'm-0 overflow-auto p-4',
+            'text-sm text-gray-800 dark:text-gray-200',
+            'font-mono',
+            'bg-white dark:bg-gray-900'
+          )}
+        >
+          {viewableRows.join('\n')}
```

Comment:
I think is a bit redundant having the same information displayed twice?
![Screenshot 2025-04-09 at 14 24 48](https://github.com/user-attachments/assets/0dd4d149-8c9d-4321-ac41-4a86272f937c)